integration-tests/src/test/resources/sitconfig/java/SitConfigTests.java

```diff
@@ -0,0 +1,554 @@
+
+  }*/
+  public SitConfigTests(String adminHost, String adminPort, String adminUser, String adminPassword)
+      throws Exception {
+    this.adminHost = adminHost;
+    this.adminPort = adminPort;
+    this.adminUser = adminUser;
+    this.adminPassword = adminPassword;
+    createConnections();
+  }
+
+  private void createConnections() throws Exception {
+    try {
+      runtimeMbs =
+          lookupMBeanServerConnection(
+              adminHost,
+              adminPort,
+              adminUser,
+              adminPassword,
+              RuntimeServiceMBean.MBEANSERVER_JNDI_NAME);
+      domainRuntimeMbs =
+          lookupMBeanServerConnection(
+              adminHost,
+              adminPort,
+              adminUser,
+              adminPassword,
+              DomainRuntimeServiceMBean.MBEANSERVER_JNDI_NAME);
+      editMbs =
+          lookupMBeanServerConnection(
+              adminHost,
+              adminPort,
+              adminUser,
+              adminPassword,
+              EditServiceMBean.MBEANSERVER_JNDI_NAME);
+
+      ObjectName runtimeserviceObjectName = new ObjectName(RuntimeServiceMBean.OBJECT_NAME);
+      runtimeServiceMBean =
+          (RuntimeServiceMBean)
+              MBeanServerInvocationHandler.newProxyInstance(runtimeMbs, runtimeserviceObjectName);
+
+      ObjectName domainServiceObjectName = new ObjectName(DomainRuntimeServiceMBean.OBJECT_NAME);
+      domainServiceMBean =
+          (DomainRuntimeServiceMBean)
+              MBeanServerInvocationHandler.newProxyInstance(
+                  domainRuntimeMbs, domainServiceObjectName);
+
+      ObjectName editserviceObjectName = new ObjectName(EditServiceMBean.OBJECT_NAME);
+      editServiceMBean =
+          (EditServiceMBean)
+              MBeanServerInvocationHandler.newProxyInstance(editMbs, editserviceObjectName);
+    } catch (MalformedObjectNameException ex) {
+      Logger.getLogger(SitConfigTests.class.getName()).log(Level.SEVERE, null, ex);
+      throw new AssertionError(ex.getMessage());
+    }
+  }
+
+  private MBeanServerConnection lookupMBeanServerConnection(
+      String host, String adminPort, String user, String adminPassword, String jndiName)
+      throws MalformedURLException, IOException, Exception {
+    JMXServiceURL serviceURL;
+    MBeanServerConnection mBeanServerConnection;
+
+    println(
+        "Host: "
+            + adminHost
+            + " Port: "
+            + adminPort
+            + " adminUser: "
+            + adminUser
+            + " adminPassword :"
+            + adminPassword);
+    String protocol = "t3";
+    Integer portInteger = Integer.valueOf(adminPort);
+    int port = portInteger;
+    HashMap h = new HashMap();
+    h.put(Context.SECURITY_PRINCIPAL, adminUser);
+    h.put(Context.SECURITY_CREDENTIALS, adminPassword);
+    h.put(JMXConnectorFactory.PROTOCOL_PROVIDER_PACKAGES, "weblogic.management.remote");
+    h.put("jmx.remote.x.request.waiting.timeout", new Long(10000));
```

Comment:
Is this timeout important or simply meant to be "forever"?

Reply:
I want the jmx client to timeout if the method invocation doesn't return in 10secs.

Reply:
Similarly to the earlier comment, what I'm really trying to figure out is if this test is of the timeout or if you believe that the test would never take 10 seconds and so this timeout is unimportant.

Reply:
Yes, this timeout is not really important to the test and just added to timeout the method invocation if there are any hangs.